06-descriptive-analysis-in-srvyr.Rmd

```diff
@@ -1 +1,420 @@
+
+Measures of central tendency find the central (or average) responses. These include the mean, median, and mode.
+
+* Examples: Average 2000 Academic Performance Index in California, the median house price in Canada
+
+The `survey_mean()` and `survey_median()` functions from {srvyr} calculate mean and median using survey data, respectively.
+
+```{r}
+# Calculate survey mean of variable api00
+
+dstrata %>%
+  group_by(awards) %>%
+  summarise(api00_mean = survey_mean(api00),
+            api00_median = survey_median(api00))
+```
+
+#### Measures of dispersion
+
+Measures of dispersion describe how data spreads around the central tendency. These measures include the range and standard deviation.
+
+* Examples: The standard deviation of the 2000 Academic Performance Index in California, the range between the minimum and maximum electricity expenditure in Ohio
+
+The `survey_var()` and `survey_sd()` functions from the {srvyr} package calculate population variance and standard deviation using survey methods, respectively.
+
+```{r}
+# Calculate the variance and standard deviation 
+dstrata %>%
+  summarise(api99_var = survey_var(api00),
+            api99_sd = survey_sd(api00))
+```
+
+#### A brief refresher on the `dplyr::summarize()` function
+<!-- TODO: Put this in a callout box -->
+Several functions in {srvyr} must be called within the `dplyr::summarize()` function. Recall that `summarize()` collapses many values down to a single summary:
+
+```{r}
+mtcars %>% 
+  summarize(mean_hp = mean(hp))
+```
+
+These verbs can be used in conjunction with `group_by()`, applying the functions on a group-by-group basis to create grouped summaries.
+
+```{r}
+mtcars %>% 
+  group_by(cyl) %>% 
+  summarize(mean_hp = mean(hp))
+```
+<!--TODO: end callout note-->
+
+### Bivariate/multivariate analysis
+
+Bivariate analysis concerns the relationship between two variables, while Multivariate analysis studies the relationship between multiple variables. The correlation is a number between -1 and +1 that denotes the strength of the linear relationship between two variables.
+
+* Examples: Relationship between academic performance scores in 1999 and 2000; relationship between age and income 
+
+The {srvyr} package does not include a function for calculating correlations for survey variables. We can use the {survey} package's `svyvar()` estimates the variance between the variables and then `cov2cor()` to get the correlation. Another option is to use the `svycor()` function from the {jtools} package.
+
+```{r}
+# Calculate correlation between API 2000 and API 1999.
+
+var <- survey::svyvar(~api00 + api99, design = dstrata)
+vmat <- as.matrix(var)
+attr(vmat, "var") <- NULL # Remove extra info
+attr(vmat, "statistic") <- NULL # Remove extra info
+cov2cor(vmat)
+
+# Alternatively
+dstrata |>
+  survey::svyvar(~api00 + api99, design = _) |>
+  (\(x) as.matrix(x))() |>
+  mutate(across(everything(), var, NULL)) |>
+  (\(x) cov2cor(x))()
```

Comment:
This code doesn't run for me. Maybe this instead or wait for srvyr to be updated... 

```suggestion
# Alternatively
dstrata |>
  survey::svyvar(~api00 + api99, design = _) |>
  (\(x) as.matrix(x))() |>
  (\(x) cov2cor(x))()
```

Reply:
This will show up when I push up my changes, but the code I have is:

```r
dstrata |>
 survey::svyvar(~api00 + api99, design = _) |>
 as.matrix(x = _) |>
 labelled::remove_attributes(x = _, "var") |>
 labelled::remove_attributes(x = _, "statistic") |>
 stats::cov2cor(V = _)
```

the output is:

```
          api00     api99
api00 1.0000000 0.9759047
api99 0.9759047 1.0000000
```

but if there's a better option, please let me know!